web/src/main/java/org/fao/geonet/proxy/URITemplateProxyServlet.java

```diff
@@ -48,4 +89,61 @@ protected HttpClient createHttpClient(RequestConfig requestConfig) {
             .useSystemProperties()
             .build();
     }
+
+    protected void service(HttpServletRequest servletRequest, HttpServletResponse servletResponse)
+        throws ServletException, IOException {
+        switch (securityMode) {
+            case NONE:
+                super.service(servletRequest, servletResponse);
+                break;
+            case DB_LINK_CHECK:
+                boolean proxyCallAllowed = false;
+
+                // Check if user is authenticated
+                UserSession userSession = ApiUtils.getUserSession(servletRequest.getSession());
+                if (userSession.isAuthenticated()) {
```

Comment:
Can `ApiUtils.getUserSession` return `null`? In that case this will throw a `NullPointerException`. Shouldn't it throw a `SecurityException` in that case?

Reply:
If the session is null, then you get 
![image](https://user-images.githubusercontent.com/1701393/82911117-c7c33e80-9f6b-11ea-9d39-b0ec53472253.png)

See
https://github.com/geonetwork/core-geonetwork/blob/master/services/src/main/java/org/fao/geonet/api/ApiUtils.java#L178-L180